clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -300,9 +301,13 @@ void runOnce() {
             try {
                 transactionManager.maybeResolveSequences();
 
+                RuntimeException lastError = transactionManager.lastError();
+                if (transactionManager.hasAbortableError() && shouldHandleAuthorizationError(lastError)) {
+                    return;
+                }
+
```

Comment:
I thought I left this comment, but seems like it didn't take -- should we move the fatal error check above the auth check? Since for other request types, we will go through the above path, but it really should just be a fatal error.

Reply:
sounds good - perhaps you meant by checking for fatal first then abortable error? Though I think it doesn't change the logic there.

Reply:
Yup -- my concern is we unecessarily reset the producer to initializing in the fatal error case.

Reply:
Ah I see we had the abortable error check. 😅 Ok well now we are doubly covered.